src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int RESIZING_CONSTANT = 2;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap(int capacity) {
+        table = new Node[capacity];
+    }
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= table.length * LOAD_FACTOR) {
+            resizeTable();
+        }
+
+        int index = (key == null) ? 0 : getIndex(key);
```

Comment:
you have the same check in the `getIndex(key)` method
```suggestion
        int index = getIndex(key);
```